montecarlo/pythia8/CMakeLists.txt

```diff
@@ -9,6 +9,9 @@
 # @author Pere Mato, CERN
 ############################################################################
 
+# required also for cling
+include_directories(${PYTHIA8_INCLUDE_DIR})
```

Comment:
Why it was necessary to regress here?

Reply:
While it needs includes for dictionary generation - see code.
It was failing on some platforms when I was testing my other PR

Reply:
So, some of the dictionaries that failed missed a dependency to pythia8?

Reply:
It is a way how current dictionary generation works.
It able to extract includes which are configured via `target_include_directories`. 
After my bigger PR it will not be a case - one needs to use `include_directories` to set includes for dictionary. So to say - this change is preparation for next PR

Reply:
> After my bigger PR it will not be a case - one needs to use include_directories to set includes for dictionary.

Can you then remove it from this PR, please? We'll discuss the requirement of using `include_directories` in the other PR.

Reply:
Currently `include_directories` and `target_include_directories` are equivalent for dictionary generation - they leads to same arguments for `cling`. In my mind, using `include_directories` express more clearly that includes required for the cling

Reply:
> In my mind, using include_directories express more clearly that includes required for the cling

That's something we should probably be discussing in the other PR? For starters, check the arguments provided here: https://pspdfkit.com/blog/2018/modern-cmake-tips/

Reply:
But `target_include_directories` does not work for dictionary target. Command we produce for the `rootcling` can not be adjusted via `target_include_directories(G__Something PRIVATE ${CUSTOM_INCLUDE_PATH})`. And `include_directories` is simple alternative - especially while our cmake files split on many subprojects

Reply:
`include_directories` is 'global' state which may break encapsulation. I'd prefer not to make this particular change as part of this PR and deal with rootcling elsewhere.

Reply:
It is global until end of current cmake file, it is not extend to other targets in other cmake files